### DISCUSSION TITLE
Multiple Prompts

### DISCUSSION THREAD

**Discussion by gebaltso** [lllyasviel/Fooocus#2978] [Q&A] [answered]
Hello! I use version 2.3.1. I followed the instructions of [https://github.com/lllyasviel/Fooocus/discussions/1751#discussion-6028822] but didn't manage to get any image. The app starts and shows nothing. When I click on Generate I get  Error : if len(task.args) == 0:
AttributeError: 'str' object has no attribute 'args'. 
Can I do something else to get images generated according to a queue of prompts? And if so, can I also create different images for the same id (having that the initial image depicts a face) giving a queue of text prompts?  
Eg wear a hat
shopping in a market
being sad
etc

**Comment by Rogala** [comment, marked as answer]
If you want to run multiple generation with different promts, use the wildcards folder, create a text file in it and write your promts each on a new line.

To start the generation in the promt input window, specify the file name with two underscores on both sides as in the example `__name__`

If you want to change part of the promt, you can use arrays by enclosing the part of the promt that is changed in square brackets followed by a comma. For example: **(don't forget to set the number of pictures that the array will produce)**

`Flower [[blue, white, green]]`
this will give you 3 images with different colors of the same flower.

You can combine the work of the array and the wildcards file to create a query of any complexity.

For example: 3 files in wildcards have different datasets, the first the name of the flowers, the second their color, the third a description of the background.

`Flower [[__file1__, __file2__, __file3__]]`
depending on the amount of data in these files you will get the number of variants `F1*F2*F3=`

**Reply by gebaltso** [reply]
Thanks a lot I tried it and works great!
Now about the id modification is there a similar to the above way to get alterations of the same image? Like use FaceSwap along with different prompts. For example for each image I created via Fooocus I want to get modified versions of it in terms of hair color. How can I achieve it in an efficient manner and not one by one? Thanks in advance!

**Reply by Rogala** [reply]
Specify in the promt.
 “Hair color [white, black, red]”
or from the wildcards folder where the color list file is located.
`“Hair color __h-color__”`

**Reply by gebaltso** [reply]
How can I do this for images that I have already though? I uploaded an image and used FaceSwap and in the prompt I used something like “Hair color __h-color__” . But do I have to do so for each image I want to create modifications? Is there any way to do it for a folder of about 100 images ? Thanks in advance and sorry for the many questions.

**Reply by Rogala** [reply]
No batch image processing
